zipkin-server/src/test/kotlin/zipkin2/server/internal/brave/ITZipkinSelfTracing.kt

```diff
@@ -28,37 +30,80 @@ import zipkin2.storage.InMemoryStorage
 @SpringBootTest(
   classes = [ZipkinServer::class],
   webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
-  properties = ["spring.config.name=zipkin-server", "zipkin.self-tracing.enabled=true"]
+  properties = [
+    "spring.config.name=zipkin-server",
+    "zipkin.self-tracing.enabled=true",
+    "zipkin.self-tracing.message-timeout=1ms",
+    "zipkin.self-tracing.traces-per-second=0"
+  ]
 )
 @RunWith(SpringRunner::class)
 class ITZipkinSelfTracing {
   @Autowired lateinit var server: Server
   @Autowired lateinit var storage: TracingStorageComponent
-  @Before fun clearStorage() = (storage.delegate as InMemoryStorage).clear()
+
+  lateinit var inMemoryStorage: InMemoryStorage
+
+  @Before fun setUp() {
+    inMemoryStorage = (storage.delegate as InMemoryStorage)
+  }
+
+  @After fun clearStorage() {
+    inMemoryStorage.clear()
+  }
 
   @Test fun getIsTraced_v2() {
     assertThat(Http.getAsString(server, "/api/v2/services"))
       .isEqualTo("[]")
 
     assertServerTraced()
+
+    val traces = inMemoryStorage.traces
+    assertThat(traces).hasSize(1)
+    assertThat(traces[0]).anyMatch {
+      it.name() == "get" && it.tags()["http.path"] == "/api/v2/services"
+    }
+  }
+
+  @Test fun staticNotTraced() {
+    // Get a couple of never-changing static resource URLs
+    assertThat(Http.getAsString(server, "/zipkin/")).contains("<html>")
+    assertThat(Http.get(server, "/zipkin/favicon.ico").isSuccessful).isTrue()
+
+    // Can just run another test as the above request should not affect its expectations.
+    getIsTraced_v2()
   }
 
   @Test fun postIsTraced_v1() {
     assertThat(Http.post(server, "/api/v1/spans", body = "[]").isSuccessful).isTrue()
 
     assertServerTraced()
+
+    val traces = inMemoryStorage.traces
+    assertThat(traces).hasSize(1)
+    assertThat(traces[0]).anyMatch {
+      it.name() == "post" && it.tags()["http.path"] == "/api/v1/spans"
+    }
   }
 
   @Test fun postIsTraced_v2() {
     assertThat(Http.post(server, "/api/v2/spans", body = "[]").isSuccessful).isTrue()
 
     assertServerTraced()
+
+    val traces = inMemoryStorage.traces
+    assertThat(traces).hasSize(1)
+    assertThat(traces[0]).anyMatch {
+      it.name() == "post" && it.tags()["http.path"] == "/api/v2/spans"
+    }
   }
 
   private fun assertServerTraced() {
-    Thread.sleep(1500) // wait for reporting interval
+    // wait for spans
+    await().untilAsserted {
+      assertThat(inMemoryStorage.acceptedSpanCount()).isGreaterThanOrEqualTo(1)
```

Comment:
specifically we should check for recursion.. like await until quiet (to ensure it will ever be quiet), but we've also never done that. So this is good in other words :)